discussions/secure_web_api.md

```diff
@@ -0,0 +1,145 @@
+
+### Provision of Certificates
+
+As mentioned before, users can use already pre-existing certificates. However, L3AF can also provide mechanisms
+(manual or auto) to generate certificates.
+
+L3AF can have a library for self-signed certificate generation, that can implement APIs with different parameters, and
+these APIs can be integrated with L3AFD for auto-generation of the certificates. In this case, L3AFD can generate
+self-signed certificates, i.e., a set of self-signed root certificates, a pair of server certificates and client
+certificates at start-up if the certificates are not found (These certificates are not recommended for production use).
+
+Also, a certificate generation tool can be implemented using this library to generate the certificates manually.
+If the user has a set of root certificates (can be from a trusted CA provider or provided by its organisation or
+self-signed) and does not have the server and client certificates, then it can use the tool provided by L3AF to generate
+the server and client certificates. The tool can also have an option to generate the root certificates and the user can
+use this option to generate self-signed root certificates manually.
+
+#### Certificate creation options
+
+To create certificates L3AFD needs following options, these options are stored in L3AFD configuration file and for
+the tool, these can be passed as parameters.
+
+- Domain (by default, localhost or FQDN)
+- Password provided by user 
+- Message Digest (sha256 or sha512 or …)
+- Encryption algorithms (Camellia, Aria and AES is the popular one)
+- Subject (Country, State, Locality, Organisation, Organisation Unit and Common Name)
+  (e.g., “/C=IN/ST=KA/L=BGL/O=WM/OU=TBT/CN=$cn”)
+- Expiration days (365, 730 or 3652 days)
+- Alternate DNS, and IP (by default localhost, and 127.0.0.1)
+
+### Location of Certificates
+
+The default location for the certificates can be ‘/etc/ssl/l3af/certs/’ and this path can also be changed through a
+l3afd configuration option in cases where the user would like to use a custom location.
+
+In case of auto-generation of the certificates, L3AFD will place the root certificates, the server certificates in the
+configured path, and the client certificates will be placed in a subdirectory ‘client’ under the configured path.
+
+In case of manually generated certificates or existing certificates, the user will have to place the root certificates
+and the server certificates in the configured directory path before starting L3AFD.
+
+### Enabling mTLS
+
+L3AFD will provide a flag to enable mTLS and this can be configured in l3afd.cfg, by default mTLS will be disabled.
+
+## Minimum TLS version
+
+TLS version popularly supported in the market v1.2 and v1.3. L3AF can support v1.3 and above.
+
+## L3AFD web Api FQDNs (Fully Qualified Domain Name)
+
+L3AFD can be configured to listen on FQDN by providing option in configuration file (i.e., l3afd.cfg) or by default,
+l3afd will listen on localhost.
```

Comment:
I think this should say L3AFD can be configured to listen on a different IP address / interface than localhost (127.0.0.1).  There could be an additional check to only accept traffic from the specified FQDN (Host header) or SNI if TLS.